### PR TITLE
 Unpin pandas version + update tag description

### PR DESCRIPTION
**Pull request by RoyStegeman**

### CODE REVIEW

**Comment by scarlehoff** [comment]
I cannot see any problems with the report for the bot.

(I've added a specific commit of curio but only for the `pyproject.toml` since we are not doing anything beyond 3.11 with conda yet while the python installation can go beyond 3.12, but it will be also a problem eventually: https://github.com/conda-forge/curio-feedstock/pull/16)

We need to move away from curio but reportengine depends on it...

**Comment by RoyStegeman** [comment]
I'm surprised the fits are not exactly the same, I directly branched this of #1959

**Comment by scarlehoff** [comment]
oh no...

**Comment by scarlehoff** [comment]
Between the two bots conda indexed the new eko which relaxes a bunch of dependencies. If you rebase it should be ok. Even just merging should be ok

**Comment by RoyStegeman** [comment]
I had a look at the package versions used by the fitbot here an in #1959, and besides the different pandas version really the only difference is that here also `python-tzdata`, a timezone package that I'm pretty sure provides functionality in pandas that we don't use anyway, is being installed. 

Maybe 2.0 has some changes to rounding/numerical stability that can explain the difference?

**Comment by scarlehoff** [comment]
We use pandas so heavily that I'm not particularly worried about this changes... as long as we consider this part of the 4.1-beta (so pandas 2 doesn't go in the tag).

**Comment by RoyStegeman** [comment]
Okay if you're happy please approve this. 

> btw, I'll add another bot so that we can have an "exact fitbot" and "last stable version fitbot"

:+1:

File: .github/workflows/fitbot.yml [hunk 1]
```diff
@@ -10,8 +10,8 @@ on:
 env:
   N3FIT_MAXNREP: 20 # total number of replicas to fit
   POSTFIT_NREP: 16 # requested replicas for postfit
-  REFERENCE_SET: NNBOT-c0f99b7b3-2024-02-28 # reference set for exact results
-  STABLE_REFERENCE_SET: NNBOT-c0f99b7b3-2024-02-28 # reference set for last tag
+  REFERENCE_SET: NNBOT-3e7c84220-2024-02-28 # reference set for exact results
+  STABLE_REFERENCE_SET: NNBOT-3e7c84220-2024-02-28 # reference set for last tag
```
Comment (scarlehoff) [inline .github/workflows/fitbot.yml:14]:
```suggestion
  STABLE_REFERENCE_SET: NNBOT-c0f99b7b3-2024-02-28 # reference set for last tag
```

For the tag we should keep the previous one!

Comment (RoyStegeman) [reply .github/workflows/fitbot.yml:14]: Ah right, I thought you wanted to tag after this PR

**Pull request merged**